Services/Devices.cs

```diff
@@ -183,13 +184,13 @@ public IDeviceClient GetClient(Device device, IoTHubProtocol protocol)
             {
                 if (e.InnerException != null && e.InnerException.GetType() == typeof(TaskCanceledException))
                 {
-                    // We get here when the cancellation token is triggered, which is fine
-                    this.log.Debug("Device creation task canceled", () => new { deviceId, e.Message });
-                    return null;
+                    var timeSpent = DateTimeOffset.UtcNow.ToUnixTimeMilliseconds() - now.ToUnixTimeMilliseconds();
+                    this.log.Error("Create device task timed out", () => new { timeSpent, deviceId, e.Message });
+                    throw;
```

Comment:
What was happening before when we didn't throw?  Did the device just sit in an indeterminate state?  ...or did returning null cause something outside of here to break?

Reply:
The `null` returned here was used as a valid device object in one of the state machines, leading to nullref exceptions later on, and bringing that SM in a non-recoverable state.